RFS/src/main/java/org/opensearch/migrations/bulkload/workcoordination/OpenSearchWorkCoordinator.java

```diff
@@ -419,36 +416,35 @@ private int numWorkItemsArePending(
                 + "      }"
                 + "    ]"
                 + "  }"
-                + "}"
+                + "},"
+                + "\"size\": 1"
                 + "}";
 
-            var path = INDEX_NAME + "/_search" + (maxItemsToCheckFor <= 0 ? "" : "?size=" + maxItemsToCheckFor);
+            var path = INDEX_NAME + "/_search";
             var response = httpClient.makeJsonRequest(AbstractedHttpClient.POST_METHOD, path, null, queryBody);
-
-            final var resultHitsUpper = objectMapper.readTree(response.getPayloadBytes()).path("hits");
             var statusCode = response.getStatusCode();
             if (statusCode != 200) {
                 throw new IllegalStateException(
-                    "Querying for pending (expired or not) work, "
-                        + "returned an unexpected status code "
-                        + statusCode
-                        + " instead of 200"
+                        "Querying for pending (expired or not) work, "
+                                + "returned an unexpected status code "
+                                + statusCode
+                                + " instead of 200"
                 );
             }
-            return resultHitsUpper.path("hits").size();
+            return objectMapper.readTree(response.getPayloadBytes()).path("hits").path("total").path("value").asInt();
         }
     }
 
     @Override
     public int numWorkItemsArePending(Supplier<IWorkCoordinationContexts.IPendingWorkItemsContext> contextSupplier)
         throws IOException, InterruptedException {
-        return numWorkItemsArePending(-1, contextSupplier);
+        return numWorkItemsArePendingInternal(contextSupplier);
```

Comment:
how do you get back the right value for this if you're running the query w/ size=1?